crates/rolldown_utils/src/dataurl.rs

```diff
@@ -16,3 +17,69 @@ pub fn encode_as_shortest_dataurl(mime: &Mime, buf: &[u8]) -> String {
 
   format!("data:{mime};{body}")
 }
+
+// Port from https://github.com/vitejs/vite/blob/main/packages/vite/src/node/plugins/dataUri.ts.
+pub fn deserialize_dataurl(s: &str) -> anyhow::Result<(Mime, Vec<u8>)> {
+  if !s.starts_with("data:") {
+    return Err(anyhow::anyhow!("Invalid dataurl"));
+  }
+  let data_url_re =
```

Comment:
Wrap it with a `std::sync::LazyLock` or the **Regex** will construct each function invocation. https://github.com/rust-lang/regex/blob/master/README.md#usage-avoid-compiling-the-same-regex-in-a-loop

Reply:
Solved.